projects/bp-gallery/src/components/common/DonateButton.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button, Dialog } from '@mui/material';
+import { PayPalButtons, PayPalScriptProvider } from '@paypal/react-paypal-js';
+import { useEffect, useRef, useState } from 'react';
+import CurrencyInput from 'react-currency-input-field';
+import './DonateButton.scss';
+import PrimaryButton from './PrimaryButton';
+import { OnApproveActions, OnApproveData } from '@paypal/paypal-js';
+import { useTranslation } from 'react-i18next';
+
+const DonateButton = ({
+  clientId,
+  donationText,
+  purposeText,
+}: {
+  clientId: string;
+  donationText?: string;
+  purposeText?: string;
+}) => {
+  const { t } = useTranslation();
+  const [isClicked, setIsClicked] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [dialogText, setDialogText] = useState('');
+  const [donation, setDonation] = useState('0.00');
+  const transitionRef = useRef(null);
+  const paypalOptions = {
+    'client-id': clientId,
+    currency: 'EUR',
+  };
+
+  const onApprove = (data: OnApproveData, actions: OnApproveActions) => {
+    return actions.order.capture().then(() => {
+      setDialogText(t('archives.edit.paypal.thankyou-label', { amount: donation }));
+      setIsDialogOpen(true);
+    });
+  };
+
+  const onError = (err: any) => {
+    setDialogText(t('archives.edit.paypal.error-label'));
+    setIsDialogOpen(true);
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      setTimeout(function () {
+        transitionRef.current.className = 'parent-active';
+      }, 20);
```

Comment:
Why wrap this in a `setTimeout`? Also, if the timeout isn't needed, you could get rid of the ref and set the class directly on the element, when `isOpen` is `true`.

Reply:
PaypalButtons creates an order on render, but we don't know the amount the user wants to donate if we render it before he is able to enter it into the textfield.

The timeout is necessary for the CSS transition. Otherwise we wouldn't have a way to set the height to 0 in the beginning and then to one and add a transition

Reply:
I don't understand what the first part of your answer has to do with my concern, sorry. My proposal doesn't change when the `PaypalButtons` are rendered, only with which class names. Maybe you can elaborate?

Regarding the second paragraph: In that case, you can use a CSS animation instead of relying on transitions. Something like this:
```scss
.paypal-parent {
  overflow: hidden;
  animation: paypal-open 1s;

  @keyframes paypal-open {
    from {
      max-height: 0px;
    }

    to {
      max-height: 60rem;
    }
  }
}
```
If you decide to use tailwind instead of scss, you can do it like this:
```html
<div classname='overflow-hidden animate-paypal-open'>
  <PayPalButtons
    ...
```
And in `tailwind.config.cjs`:
```ts
extend: {
  ...
  animation: {
    'paypal-open': 'paypal-open 1s',
  },
  keyframes: {
    'paypal-open': {
      from: {
        maxHeight: '0px',
      },
      to: {
        maxHeight: '60rem',
      },
    },
  },
},
```
While you're at it, maybe change the `60rem` to something smaller, like `20rem`? That makes the animation play out more smoothly. Alternatively, you could change the animation time :)